src/backend/distributed/utils/node_metadata.c

```diff
@@ -1525,26 +1525,27 @@ TupleToWorkerNode(TupleDesc tupleDescriptor, HeapTuple heapTuple)
 	WorkerNode *workerNode = NULL;
 	bool isNull = false;
 
-	Datum nodeId = heap_getattr(heapTuple, Anum_pg_dist_node_nodeid,
+	HeapTuple expandedTuple = ExpandTupleIfNeeded(heapTuple, tupleDescriptor);
+	Datum nodeId = heap_getattr(expandedTuple, Anum_pg_dist_node_nodeid,
```

Comment:
I think this is the wrong fix. I think you rather should just deform the tuple and access the columns via that. That'll take care of the expansion *and* be faster.

Reply:
Makes sense. Just for my understanding, is it just that tuple deformation is the recommended way, or are there any problems with this approach?